app/controllers/teachers_controller.rb

```diff
@@ -20,41 +22,26 @@ def create
     @school = School.new(school_params)
     # Find by email, but allow updating other info.
     @teacher = Teacher.find_by(email: teacher_params[:email])
-    if @teacher
-      # Exclude website from the school finder, so we can update an existing school.
-      @school = School.find_or_create_by(name: school_params[:name], city: school_params[:city], state: school_params[:state])
-      @school.website = school_params[:website]
-      @school.save!
-      @teacher.update(teacher_params)
-      @teacher.school = @school
-      @teacher.save!
-      if @teacher.application_status == "Validated"
-        TeacherMailer.welcome_email(@teacher).deliver_now
-        TeacherMailer.form_submission(@teacher).deliver_now
-        TeacherMailer.teals_confirmation_email(@teacher).deliver_now
-        flash[:success] = "Thanks! We have updated your information. We have sent BJC info to #{@teacher.email}."
-      else
-        flash[:success] = "Thanks! We have updated your information."
-      end
+    if @teacher and defined?(current_user.id) and current_user.id == @teacher.id
```

Comment:
This line might benefit from the safe navigator option in Ruby:
`if @teacher and current_user&.id == @teacher.id`
[See this commentary](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) for details, as well as the [shorthand block on codeclimate about cognitive complexity](https://docs.codeclimate.com/docs/cognitive-complexity), it basically checks if each element before &. is nil.

Very minor concern, if codeclimate isn't saying anything about cognitive complexity, no need to change anything.